great_expectations/_docs_decorators.py

```diff
@@ -60,6 +71,24 @@ def add(self, func: F) -> None:
             logger.exception(f"Could not add this function to the public API list: {func}")
             raise
 
+    def _add_to_class_registry(self, func: F) -> None:
+        if isinstance(func, type):
+            key = f"{func.__module__}.{func.__qualname__}"
+            self._class_registry[key].add(self.CLASS_DEFINITION)
+        else:
+            parts = func.__qualname__.split(".")
+            METHOD_PARTS_LENGTH = 2  # Standalone functions will have a length of 1
+            if len(parts) >= METHOD_PARTS_LENGTH:
+                cls = parts[0]
+                method = parts[1]
```

Comment:
When len(parts) > 2 do we still want to set the method to parts[1]? Is it ever > 2? If not, I think we should raise when this is > 2 which an error message to update this method to handle that case.

Reply:
An example is: `DataSourceManager._register_add_datasource.<locals>.crud_method_info`

This happens when we have closures